### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DavidGlusch**

### CODE REVIEW

File: app/test_main.py [hunk 1]
```diff
@@ -1,3 +1,60 @@
+import pytest
+
 from app.main import get_human_age
 
-# write your code here
+
+class TestGetHumanAge:
+    @pytest.mark.parametrize(
+        "cat_age,dog_age,human_years",
```
Comment (WDemidenko) [inline app/test_main.py:8]: You can replace it with one tuple of ages, and then unpack it inside your function in test as arguments

**Review by WDemidenko — APPROVED** [review]
Good job!

File: app/test_main.py [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+
+    @pytest.mark.parametrize(
+        "cat_age,dog_age,expected_error",
+        [
+            pytest.param(
+                "28", "28", TypeError,
+                id="expected to raise TypeError when not int or float passed"
+            )
+        ]
+    )
```
Comment (YuriiKindrat) [inline app/test_main.py:52]: no need to use mark.parametrize for just one case, use simple test for this

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]

File: app/test_main.py [hunk 3]
```diff
@@ -1,3 +1,52 @@
...
+            dog_age: int,
+            human_years: list
+    ) -> None:
+
+        if human_years == TypeError:
+            with pytest.raises(TypeError):
+                get_human_age(cat_age, dog_age)
+
+        else:
+            assert get_human_age(cat_age, dog_age) == human_years
```
Comment (Serhii-Leonenko) [inline app/test_main.py:52]: You should split the test which checks the data type, but as Yurii said, you can make it without pytest.mark.parametrize

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

File: app/test_main.py [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+            pytest.param(
+                1256, 1256, [310, 248],
+                id="should work properly with big numbers"
+            ),
+            pytest.param(
+                25.2, 25.2, [2.0, 2.0],
+                id="should work properly with float values"
+            )
+        ]
+    )
```
Comment (Serhii-Leonenko) [inline app/test_main.py:35]: You should add more tests for all age boundaries, for example, you check here that 14, 14 gives 0, 0 in the result, but you don't check that 15, 15 gives 1, 1 in the result, etc.

image

**Review by Serhii-Leonenko — CHANGES_REQUESTED** [review]

**Review by YuriiKindrat — APPROVED** [review]
